client/src/unifycr-sysio.c

```diff
@@ -828,8 +828,12 @@ ssize_t UNIFYCR_WRAP(write)(int fd, const void *buf, size_t count)
                 return (ssize_t) (-1);
             }
         } else {
+            ssize_t rc;
+
             fd += unifycr_fd_limit;
-            pwrite(fd, buf, count, filedesc->pos);
+            rc = pwrite(fd, buf, count, filedesc->pos);
+            if (rc < 0)
+                perror("pwrite failed");
         }
```

Comment:
Here we should set errno appropriately and return (ssize_t)-1 to the user and let them handle it, similar to what we do higher up in this routine:

            /* ERROR: invalid file descriptor */
            errno = EBADF;
            return (ssize_t) (-1);

Reading the man page for pwrite, it sets errno, so we just need to return (ssize_t) -1 and perhaps make a comment that pwrite sets errno for us.

Reply:
Sounds good. I'll also update it to handle the short-write case where pwrite() returns < count.